include/simdjson/generic/ondemand/array-inl.h

```diff
@@ -71,11 +71,16 @@ simdjson_really_inline simdjson_result<array_iterator> array::begin() noexcept {
 simdjson_really_inline simdjson_result<array_iterator> array::end() noexcept {
   return array_iterator(iter);
 }
+simdjson_really_inline error_code array::consume() noexcept {
+  auto error = iter.json_iter().skip_child(iter.depth()-1);
+  if(error) { iter.abandon(); }
+  return error;
+}
 
-simdjson_really_inline simdjson_result<std::string_view> array::raw_json_token() noexcept {
+simdjson_really_inline simdjson_result<std::string_view> array::raw_json() noexcept {
   const uint8_t * starting_point{iter.peek_start()};
-  for(simdjson_unused auto v : *this) {}
-  if(iter.error()) { return iter.error(); }
+  auto error = consume();
```

Comment:
Hmm, this probably does make it faster (we validate less, but I think this is within the principle of validating what you use).